cpp/test/Slice/errorDetection/NotType.err

```diff
@@ -6,6 +6,6 @@ NotType.ice:16: 'E' is an exception, which cannot be used as a type
 NotType.ice:17: 'Mod' is not a type
 NotType.ice:17: 'E' is an exception, which cannot be used as a type
 NotType.ice:22: 'E' is an exception, which cannot be used as a type
-NotType.ice:23: struct 'S' must have at least one member
+NotType.ice:27: 'E' is an exception, which cannot be used as a type
```

Comment:
Another thing this PR fixes, the `struct 'S' must have at least one member` was bogus.

If a field's type didn't exist, the syntax error caused us to skip over the field completely.
Then, when the parser recovered, `S` looked like an empty struct.
```
struct S
{
    FakeType foo;
}
```
But this is obviously bogus. `S is empty` is not the real error here.
The real error is that `FakeType` wasn't a real type.

This has now been fixed.